File: BannerAdExample/ObjcImplementation/ObjcImplementation/ViewController.m
//
//  ViewController.m
//  ObjcImplementation
//
//  Created by Rishabh Tripathi on 12/06/25.
//

#import "ViewController.h"

@interface ViewController () <ApBannerListener>

@property (weak, nonatomic) IBOutlet GAMBannerView *bannerView;
@property (strong, nonatomic) ApBannerAd *apBanner;

@end

@implementation ViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.apBanner = [[ApBannerAd alloc] initWithApPlacementId:@"testBannerAdUnit"];
}

-(IBAction)showAd:(UIButton *)sender {
    if (self.apBanner) {
        UIView *adView = [self.apBanner getAdView];
        [_bannerView addSubview:adView];
        adView.translatesAutoresizingMaskIntoConstraints = NO;

        [NSLayoutConstraint activateConstraints:@[
            [adView.leadingAnchor constraintEqualToAnchor:_bannerView.safeAreaLayoutGuide.leadingAnchor],
            [adView.trailingAnchor constraintEqualToAnchor:_bannerView.safeAreaLayoutGuide.trailingAnchor],
            [adView.topAnchor constraintEqualToAnchor:_bannerView.safeAreaLayoutGuide.topAnchor],
            [adView.bottomAnchor constraintEqualToAnchor:_bannerView.safeAreaLayoutGuide.bottomAnchor]
        ]];
    }
}

-(IBAction)loadAd:(UIButton *)sender {
    if (self.apBanner) {
        [self.apBanner loadAdWithViewController:self apBannerListener:self];
    }
}


@end

File: BannerAdExample/SwiftImplementation/SwiftImplementation/ViewController.swift
//
//  ViewController.swift
//  SwiftImplementation
//
//  Created by Rishabh Tripathi on 10/06/25.
//

import UIKit
import ApMobileSDK
import GoogleMobileAds

class ViewController: UIViewController, ApBannerListener {
    
    @IBOutlet weak var bannerAdView: GAMBannerView!
    var apBanner = ApBannerAd(apPlacementId: "testBannerAdUnit")

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func loadAd(_ sender: UIButton) {
        apBanner.loadAd(viewController: self, apBannerListener: self)
    }
    
    @IBAction func showAd(_ sender: UIButton) {
        let adView = apBanner.getAdView()
        bannerAdView.addSubview(adView)
        NSLayoutConstraint.activate([
            adView.leadingAnchor.constraint(equalTo: bannerAdView.safeAreaLayoutGuide.leadingAnchor),
            adView.trailingAnchor.constraint(equalTo: bannerAdView.safeAreaLayoutGuide.trailingAnchor),
            adView.topAnchor.constraint(equalTo: bannerAdView.safeAreaLayoutGuide.topAnchor),
            adView.bottomAnchor.constraint(equalTo: bannerAdView.safeAreaLayoutGuide.bottomAnchor)
        ])
    }

}

